Packs/ApiModules/Scripts/AnsibleApiModule/AnsibleApiModule.py

```diff
@@ -0,0 +1,296 @@
+
+    module_args = ""
+    # build module args list
+    for arg_key, arg_value in args.items():
+        # skip hardcoded host arg, as it doesn't related to module
+        if arg_key == 'host':
+            continue
+
+        module_args += "%s=\"%s\" " % (arg_key, arg_value)
+
+        # If this isn't host based, then all the integration parms will be used as command args
+    if host_type == 'local':
+        for arg_key, arg_value in int_params.items():
+            module_args += "%s=\"%s\" " % (arg_key, arg_value)
+
+    r = ansible_runner.run(inventory=inventory, host_pattern='all', module=command, quiet=True,
+                           omit_event_data=True, ssh_key=sshkey, module_args=module_args, forks=fork_count)
+
+    results = []
+    for each_host_event in r.events:
+        # Troubleshooting
+        # demisto.log("%s: %s\n" % (each_host_event['event'], each_host_event))
+        if each_host_event['event'] in ["runner_on_ok", "runner_on_unreachable", "runner_on_failed"]:
+
+            # parse results
+
+            result = json.loads('{' + each_host_event['stdout'].split('{', 1)[1])
+            host = each_host_event['stdout'].split('|', 1)[0].strip()
+            status = each_host_event['stdout'].replace('=>', '|').split('|', 3)[1]
+
+            # if successful build outputs
+            if each_host_event['event'] == "runner_on_ok":
+                if 'fact' in command:
+                    result = result['ansible_facts']
+                else:
+                    if result.get(command) is not None:
+                        result = result[command]
+                    else:
+                        result.pop("ansible_facts", None)
+
+                result = rec_ansible_key_strip(result)
+
+                if host != "localhost":
+                    readable_output += "# %s - %s\n" % (host, status)
+                else:
+                    # This is integration is not host based
+                    readable_output += "# %s\n" % status
+
+                readable_output += dict2md(result)
+
+                # add host and status to result if it is a dict. Some ansible modules return a list
+                if type(result) == dict:
+                    result['host'] = host
+                    result['status'] = status.strip()
+
+                results.append(result)
+            if each_host_event['event'] == "runner_on_unreachable":
+                msg = "Host %s unreachable\nError Details: %s" % (host, result)
+
+            if each_host_event['event'] == "runner_on_failed":
+                msg = "Host %s failed running command\nError Details: %s" % (host, result)
+
+            if each_host_event['event'] in ["runner_on_failed", "runner_on_unreachable"]:
+                return_error(msg)
+    return CommandResults(
+        readable_output=readable_output,
+        outputs_prefix=integration_name + '.' + command,
```

Comment:
```suggestion
        outputs_prefix=integration_name,
```
Was there a reason to have the command here? it seems to be redundent.

Reply:
As discussed, there is no guarantee that the outputs from different Ansible modules would share the same structure. As a result I prefer to isolate the different command context outputs.